### DISCUSSION TITLE
Help needed adding new Chain in AppKit Unity

### DISCUSSION THREAD

**Discussion by felixnomina** [reown-com/appkit#3298] [Q&A]
Hi, I am trying to add bsc chain in the supported chain in AppKitConfig, but it always return an error in the webgl build. Maybe I was doing this wrong or missing something?

Here is the error:

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'id')
    at Reown.framework.js:2195:45
    at Array.reduce (<anonymous>)
    at Reown.framework.js:2194:38

![image](https://github.com/user-attachments/assets/140253fa-2bbc-4eed-8a1f-50477b294367)

Here is the code that cause error

```supportedChains = new[]
{
		ChainConstants.Chains.Base,

		new Chain(ChainConstants.Namespaces.Evm,
			chainReference: "56",
			name: "BNB Smart Chain",
			nativeCurrency: new Currency("BNB", "BNB", 18),
			blockExplorer: new BlockExplorer("BscScan", "https://bscscan.com"),
			rpcUrl: "wss://bsc-rpc.publicnode.com",
			isTestnet: false,
			imageUrl: "https://cryptologos.cc/logos/binance-coin-bnb-logo.png",
			viemName: "BNB" // required only on WebGL
		)
}
```

**Comment by datphamhaui** [comment]
https://github.com/orgs/reown-com/discussions/3411#discussioncomment-11504941
You can refer to my solution
